src/BlockStorage/v2/Models/Volume.php

```diff
@@ -156,4 +159,42 @@ public function parseMetadata(ResponseInterface $response): array
 
         return isset($json['metadata']) ? $json['metadata'] : [];
     }
+
+    /**
+     * Update the bootable status for a volume, mark it as a bootable volume.
+     *
+     * @param bool $bootable
+     */
+    public function setBootable(bool $bootable)
+    {
+        $bootable = boolval($bootable);
+        $this->execute($this->api->postVolumeBootable(), ['id' => $this->id, 'bootable' => $bootable]);
+    }
+
+    /**
+     * Sets the image metadata for a volume.
+     *
+     * @param array $metadata
+     */
+    public function setImageMetadata(array $metadata)
+    {
+        $this->execute($this->api->postImageMetadata(), ['id' => $this->id, 'metadata' => $metadata]);
+    }
+
+    /**
+     * Administrator only. Resets the status, attach status, and migration status for a volume. Specify the os-reset_status action in the request body.
+     *
+     * @param array $options
+     *
+     * $options['status']             = (string)    The volume status.
+     * $options['migrationStatus']   = (string)    The volume migration status.
+     * $options['attachStatus']      = (string)    The volume attach status.      [OPTIONAL]
+     *
+     * @see https://developer.openstack.org/api-ref/block-storage/v2/index.html#volume-actions-volumes-action
+     */
+    public function resetStatus(array $options)
+    {
+        $options += ['id' => $this->id];
```

Comment:
Might prefer `array_merge` here. If they pass in an `id` this isnt going to overwrite it (Maybe you want that?).